sales_team_security/models/res_partner.py

```diff
@@ -40,3 +41,54 @@ def _onchange_parent_id_sales_team_security(self):
         """
         if self.parent_id and self.parent_id.team_id and not self.team_id:
             self.team_id = self.parent_id.team_id.id
+
+    @api.depends('company_type', 'child_ids', 'child_ids.user_id')
+    def sync_followers_in_childs(self):
+        for item in self.filtered(lambda x: x.company_type == 'company'):
+            partners = self.env['res.partner'].search([
+                ('id', 'in', (item.child_ids.ids + item.ids))
+            ])
+            user_ids = partners.mapped('user_id')
+            if user_ids:
+                partners.message_subscribe(
+                    partner_ids=user_ids.mapped('partner_id').ids
+                )
+
+    @api.depends('parent_id', 'child_ids')
+    def change_user_id_sales_team_security(self):
+        for item in self:
+            partner = item.get_partner_to_sync()
+            if partner:
+                partner.sync_followers_in_childs()
+
+    @api.depends('company_type', 'parent_id', 'child_ids')
+    def get_partner_to_sync(self):
+        if self.parent_id or self.child_ids:
+            partner = self
+            while partner.parent_id:
+                partner = partner.parent_id
+            if partner.company_type != 'company':
+                return False
+            return partner
+
+    @api.multi
+    def write(self, vals):
+        """
+        1- Remove previous_user_id as followers to all child records
+        (Prevent drash followers)
+        2- Call change_user_id_sales_team_security function to
+        sync followers
+        """
+        for item in self:
+            if 'user_id' in vals and item.user_id:
+                partner_to_sync = item.get_partner_to_sync()
+                if partner_to_sync:
+                    ids = (partner_to_sync.child_ids.ids + partner_to_sync.ids)
+                    partners = self.env['res.partner'].search([('id', 'in', ids)])
+                    partners.message_unsubscribe(
```

Comment:
Why this code is not in `sync` method? If you call it sync, do the real sync, not only half of the task.